src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -672,11 +708,50 @@ std::unique_ptr<Object::DeleteOp> D4NFilterObject::get_delete_op()
 
 int D4NFilterObject::D4NFilterReadOp::prepare(optional_yield y, const DoutPrefixProvider* dpp)
 {
+  ldpp_dout(dpp, 0) << "D4NFilterObject::" << __func__ << "()" << dendl;
+  auto blockDir = source->driver->get_block_dir(); 
+  rgw::d4n::CacheBlock block;
+  block.cacheObj.objName = source->get_name(); // for simple ops, the name will be equivalent to the oid, so we want to use the name to retrieve the block
+  block.cacheObj.bucketName = source->get_bucket()->get_name();
+  block.blockID = 0;
+  block.size = 0;
+
+  int ret = -1;
+  if ((ret = blockDir->get(dpp, &block, y)) < 0) { 
```

Comment:
this is one get() call here and then later get_obj_attrs_from_cache() invoked two more, even in cases where there are no delete markers. This needs to be modified. I think we need to get on a call to discuss this